tests/functional/adapter/files.py

```diff
@@ -0,0 +1,208 @@
+select * from (
+    select 1 as id
+) as my_subquery
+where id = 1
+"""
+
+test_ephemeral_passing_sql = """
+with my_other_cool_cte as (
+    select id, name from {{ ref('ephemeral') }}
+    where id > 1000
+)
+select name, id from my_other_cool_cte
+"""
+
+test_ephemeral_failing_sql = """
+with my_other_cool_cte as (
+    select id, name from {{ ref('ephemeral') }}
+    where id < 1000
+)
+select name, id from my_other_cool_cte
+"""
+
+model_incremental = """
+select * from {{ source('raw', 'seed') }}
+{% if is_incremental() %}
+where id > (select max(id) from {{ this }})
+{% endif %}
+""".strip()
+
+cc_all_snapshot_sql = """
+{% snapshot cc_all_snapshot %}
+    {{ config(
+        check_cols='all', unique_key='id', strategy='check',
+        target_database=database, target_schema=schema
+    ) }}
+    select * from {{ ref(var('seed_name', 'base')) }}
+{% endsnapshot %}
+""".strip()
+
+cc_name_snapshot_sql = """
+{% snapshot cc_name_snapshot %}
+    {{ config(
+        check_cols=['name'], unique_key='id', strategy='check',
+        target_database=database, target_schema=schema
+    ) }}
+    select * from {{ ref(var('seed_name', 'base')) }}
+{% endsnapshot %}
+""".strip()
+
+cc_date_snapshot_sql = """
+{% snapshot cc_date_snapshot %}
+    {{ config(
+        check_cols=['some_date'], unique_key='id', strategy='check',
+        target_database=database, target_schema=schema
+    ) }}
+    select * from {{ ref(var('seed_name', 'base')) }}
+{% endsnapshot %}
+""".strip()
+
+ts_snapshot_sql = """
+{% snapshot ts_snapshot %}
+    {{ config(
+        strategy='timestamp',
+        unique_key='id',
+        updated_at='some_date',
+        target_database=database,
+        target_schema=schema,
+    )}}
+    select * from {{ ref(var('seed_name', 'base')) }}
+{% endsnapshot %}
+""".strip()
+
+model_ephemeral_with_cte = """
+with my_cool_cte as (
+  select name, id from {{ ref('base') }}
+)
+select id, name from my_cool_cte where id is not null
+"""
+
+materialized_table = """
+  {{ config(materialized="table") }}
+"""
+
+materialized_view = """
+  {{ config(materialized="view") }}
+"""
+
+materialized_ephemeral = """
+  {{ config(materialized="ephemeral") }}
+"""
+
+materialized_incremental = """
+  {{ config(materialized="incremental") }}
+"""
+
+materialized_var = """
+  {{ config(materialized=var("materialized_var", "table"))}}
+"""
+
+model_base = """
+  select * from {{ source('raw', 'seed') }}
+"""
+
+model_ephemeral = """
+  select * from {{ ref('ephemeral') }}
+"""
+
+base_materialized_var_sql = materialized_var + model_base
+base_table_sql = materialized_table + model_base
+base_view_sql = materialized_view + model_base
+base_ephemeral_sql = materialized_ephemeral + model_base
+ephemeral_with_cte_sql = materialized_ephemeral + model_ephemeral_with_cte
+ephemeral_view_sql = materialized_view + model_ephemeral
+ephemeral_table_sql = materialized_table + model_ephemeral
+incremental_sql = materialized_incremental + model_incremental
```

Comment:
nit: Can the config vars be prefixed with `config`?  It took me a minute to see what was being combined here.  Naming with `config` helps point that out.  Plus if we do something like this in other tests where config varies (but not just by materializations) we can name the same way.

Reply:
Yeah, good idea. These files were pretty confusing and adding config_ does help.